### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by holmesie**
Added eslint to cqm-models; Fixed eslint issues

Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-32
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @okeefm
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases N/A
- [x] You have tried to break the code

**Reviewer 2:**

Name: @ljades
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "airbnb-base",
+  "rules": {
+    "max-len": ["error", {"code": 200, "ignoreComments": true}],
+    "no-unused-vars": "off",
```
Comment (okeefm) [inline .eslintrc.json]: Were there specific places that having unused variables was desired? I feel like this may be a bad rule to disable.

Comment (ljades) [reply .eslintrc.json]: I just checked. Seems it was yelling at him whenever the automatically generated includes/requires for each file were not always used for every model.

The rule definitely should be there in that case. Otherwise we'd have to somehow tailor each's includes/requires around the specific model.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "camelcase": "off",
```
Comment (ljades) [inline .eslintrc.json]: If this is the case for model fields across the board I'm going to have to change a lot of things in my current work. Should be too much effort--tracking them down though will be tough.

Comment (okeefm) [reply .eslintrc.json]: I think, but am not sure, that this is because the mongo identifiers tend to be `snake_case` rather than `camelCase`. Otherwise `camelCase` is definitely the JS convention, I believe

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "class-methods-use-this": "off"
```
Comment (ljades) [inline .eslintrc.json]: In the 3 offenses I found, all of them can be resolved by making those functions `static`.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "airbnb-base",
+  "rules": {
+    "max-len": ["error", {"code": 200, "ignoreComments": true}],
```
Comment (ljades) [inline .eslintrc.json]: I have no problem with the line limit here being 200 because of the code generation and trials regarding that. That being said, we should probably keep the 100 line limit on `js-ecqm-engine`'s end, correct?

File: .eslintrc.json (hunk 1 above)
Comment (holmesie) [reply .eslintrc.json]: Exactly right, we have imports in each model that it was complaining about. I figured it was reasonable to disable it for just this repo though, since it's only dealing with models, but I'm open to debate about it.

File: .eslintrc.json (hunk 2 above)
Comment (holmesie) [reply .eslintrc.json]: Yes, the models use `snake_case` for attributes (a trait from mongo and the back end). Since those need to be consistent, I disabled this eslint check, so it wouldn't complain about our models.

File: .eslintrc.json (hunk 4 above)
Comment (holmesie) [reply .eslintrc.json]: Yes, we should keep it to 100 in `js-ecqm-engine`. I upped it here because the model definitions can get tricky (and, like you said, how the code generation works).

File: .eslintrc.json (hunk 2 above)
Comment (ljades) [reply .eslintrc.json]: Cool. Went back and I only made a few easily trackable changes to be reverted (which I did), so we're good.

File: .eslintrc.json (hunk 3 above)
Comment (holmesie) [reply .eslintrc.json]: I cautioned against that because those basetypes were constructed directly from the mongoose documentation (which didn't use static). On second review, I think actually this is probably the way to go, and the mongoose documentation probably should of done it that way in the first place. Fixing now.

File: .eslintrc.json (hunk 1 above)
Comment (okeefm) [reply .eslintrc.json]: There are ways to tell eslint to ignore variable names that match a regex. I wonder if it'd be easy enough to write one that matches those imports, and only those?

Comment (okeefm) [reply .eslintrc.json]: If not, I'm fine with leaving it disabled.

**Comment by okeefm** [comment]
When we merge this, we may want to consider "squash and merge", or a manual squash to remove some of the commits.

File: .eslintrc.json (hunk 1 above)
Comment (holmesie) [reply .eslintrc.json]: Yeah, that is probably the way to go. I've just pushed something that re-enabled this check, but ignores those declarations.

**Comment by holmesie** [comment]
I'm a fan of committing as often as possible (each commit represents a single thought/effort), and then squashing everything on merge (the thought train of commits make up a single feature/change/bug fix/etc, contained in one PR). That way, when reviewing, it's easy to see how the PR evolved over time, but is still cleaned up on merge. Also helps to see the activity of other team members.
